pep/app/pods/components/checkbox-input/component.ts

```diff
@@ -37,7 +37,8 @@ export default class CheckboxInput extends Component<CheckboxInputArgs> {
         if (this.stopPropagation) {
             event.stopPropagation();
         }
-
-        this.args.onChange?.(!this.args.checked, event);
+        // Always get the current checked value and send it as an argument for the `onChange` event
+        const target = event.target as HTMLInputElement;
```

Comment:
you can more specifically type the `event` param so you don't need this, using our `HTMLElementEvent` type:

```ts
onChange(event: HTMLElementEvent<HTMLInputElement>) {
    //...
    this.args.onChange?.(event.target.checked, event);
}
```